extensions/src/main/java/org/pentaho/platform/osgi/KarafBoot.java

```diff
@@ -434,30 +427,45 @@ boolean canOpenConfigPropertiesForEdit( String directory ) {
     return true;
   }
 
-  void expandSystemPackages( String s ) {
+  void expandSystemPackages( String root ) {
+    Properties properties = getCustomProperties( root );
+
+    if ( properties.isEmpty() ) {
+      logger.error( "Not able to expand system.packages.extra properties due to an error loading custom.properties" );
 
-    File customFile = new File( s );
-    if ( !customFile.exists() ) {
-      logger.warn( "No custom.properties file for in karaf distribution." );
       return;
     }
-    Properties properties = new Properties();
-    FileInputStream inStream = null;
-    try {
-      inStream = new FileInputStream( customFile );
-      properties.load( inStream );
-    } catch ( IOException e ) {
-      logger
-          .error( "Not able to expand system.packages.extra properties due to an error loading custom.properties", e );
-      return;
-    } finally {
-      IOUtils.closeQuietly( inStream );
-    }
 
     properties = new SystemPackageExtrapolator().expandProperties( properties );
     System.setProperty( ORG_OSGI_FRAMEWORK_SYSTEM_PACKAGES_EXTRA,
         properties.getProperty( ORG_OSGI_FRAMEWORK_SYSTEM_PACKAGES_EXTRA ) );
+  }
+
+  private Properties getCustomProperties( String root ) {
+    Properties properties = new Properties();
+
+    final String customLocation = root + "/etc/custom.properties";
+    File customFile = new File( customLocation );
+    if ( !customFile.exists() ) {
+      logger.warn( "No custom.properties file for in karaf distribution." );
+
+      return properties;
+    }
```

Comment:
Returning an empty `Properties` object causes logging an error in `expandSystemPackages`.